kedro/runner/runner.py

```diff
@@ -74,14 +74,35 @@ def run(
         hook_manager = hook_manager or _NullPluginManager()
         catalog = catalog.shallow_copy()
 
-        unsatisfied = pipeline.inputs() - set(catalog.list())
+        # Check which datasets used in the pipeline aren't in the catalog and don't match
+        # a pattern in the catalog
+        unregistered_ds = [
+            ds
+            for ds in pipeline.data_sets()
+            if not catalog.exists_in_catalog_config(ds)
+        ]
```

Comment:
Or code for the positive instead of the negative so to create your equivalent of `catalog.list()` (named as either `catalog_list` or `registered_ds`) and not needing to change almost anything in the runner code:
```suggestion
        registered_ds = [ds for ds in pipeline.data_sets() if ds in catalog]
```